src/Confluent.Kafka/PartitionMetadata.cs

```diff
@@ -27,6 +27,24 @@ namespace Confluent.Kafka
     /// </summary>
     public class PartitionMetadata
     {
+        /// <summary>
+        ///     Initializes a new PartitionMetadata instance.
+        /// </summary>
+        /// <param name="partitionId">
+        ///     The id of the partition this metadata relates to.
+        /// </param>
+        /// <param name="leader">
+        ///     The id of the broker that is the leader for the partition.
+        /// </param>
+        /// <param name="replicas">
+        ///     The ids of all brokers that contain replicas of the partition.
+        /// </param>
+        /// <param name="inSyncReplicas">
+        ///     The ids of all brokers that contain in-sync replicas of the partition.
```

Comment:
Make a note that this value is cached by the broker and can thus be outdated and not be trusted.